### PR TITLE
Add `compute_hash` overloads for `std::byte*` to maintain backward compatibility

### PR DESCRIPTION
**Pull request by PointKernel**
Unblock https://github.com/rapidsai/cugraph/pull/4686

The `compute_hash` function is a host-device function, so the appropriate argument type is `cuda::std::byte`. However, to ensure backward compatibility and avoid breaking existing use cases, this PR adds overloads that accept `std::byte`.

### CODE REVIEW

**Review by sleeepyjack — APPROVED** [review]
The PR itself LGTM and can be merged to unblock cugraph.

However, I find this workaround a bit nasty given that `std::byte` and `cuda::std::byte` are virtually the same thing. In my book, there should exist at least an implicit conversion between the two.

Also, I think cugraph should use `cuda::std::byte` here in the first place.

**Comment by PointKernel** [comment]
> Also, I think cugraph should use cuda::std::byte here in the first place.

Agreed. I will open a cugraph PR for this once the current work is merged

**Comment by PointKernel** [comment]
> Also, I think cugraph should use cuda::std::byte here in the first place.

Opened https://github.com/rapidsai/cugraph/pull/4707 to fix this

**Pull request merged**
